code/modules/projectiles/gun.dm

```diff
@@ -179,6 +198,9 @@
 		var/acc = firemode.accuracy[min(i, firemode.accuracy.len)]
 		var/disp = firemode.dispersion[min(i, firemode.dispersion.len)]
 		process_accuracy(projectile, user, target, acc, disp)
+		
+		if(istype (src, /obj/item/weapon/gun/energy))

```

Comment:
Set a var at L154 and check it here, saves checking the type every single loop iteration.